ledger-api/grpc-definitions/com/digitalasset/admin/api/v1/package_management_service.proto

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.digitalasset.admin.api.v1;
+
+option java_outer_classname = "PackageManagementServiceOuterClass";
+option java_package = "com.digitalasset.amin.api.v1";
+
+// Allows clients to query the DAML-LF packages that are supported by the server.
+service PackageManagementService {
+
+  // Returns the details of all supported packages.
+  rpc ListPackageDetails (ListPackageDetailsRequest) returns (ListPackageDetailsResponse);
+
+  // Uploads a DAR archive.
+  rpc UploadArchive (UploadArchiveRequest) returns (UploadArchiveResponse);
+
+}
+
+message ListPackageDetailsRequest {
+
+  // Must correspond to the ledger ID reported by the Ledger Identification Service.
+  // Required
+  string ledger_id = 1;
+}
+
+message ListPackageDetailsResponse {
+
+  // The IDs of all DAML-LF packages supported by the server.
+  // Required
+  repeated string PackageDetails = 1;
+}
+
+message PackageDetails {
+ 
+  // The IDs of all DAML-LF packages supported by the server.
+  // Required
+  string package_id = 1;
+
+  // Human readable name of the package.
+  // Optional
+  string package_name = 2;
+
+  // Identifier of the agent that uploaded the package. Depending on the ledger 
+  // implementation, the interpretation of this field may differ. It could be
+  // the identifier of the uploading application or its user. It could also
+  // be the unique identifier of the participant node within the cluster.
+  // Optional
+  string uploaded_by = 3;
+
+  // Microseconds since the UNIX epoch. Can go backwards. Fixed
+  // since the vast majority of values will be greater than
+  // 2^28, since currently the number of microseconds since the
+  // epoch is greater than that. Range: 0001-01-01T00:00:00Z to
+  // 9999-12-31T23:59:59.999999Z, so that we can convert to/from
+  // https://www.ietf.org/rfc/rfc3339.txt.
+  // Required
+  google.protobuf.Timestamp upload_timestamp = 4;
+
+  // Size of the package in bytes.
+  // Required
+  uint64 package_size = 5 [jstype = JS_STRING];
+
+  // names of the modules contained in the package
+  // Optional
+  repeated string module_names = 6;
```

Comment:
Why is this optional?

Reply:
What do we need these for? If we have this, why don't we also have the identifiers within modules?

Reply:
This was raised in discussions around design of a UX for for this api. Apparently it is rather hard to understand what the package is about at the moment. Forget the lack of human readable name, which we should fix. Knowing what's inside would really help to distinguish between them. But we can punt it and start with less.